### PR TITLE
[DNM] add remove add testcase

### PR DESCRIPTION
**Pull request by tiancaiamao**
bug reproduce for https://github.com/pingcap/tikv/issues/528
don't merge it until we solve the problem

### CODE REVIEW

File: tests/raftstore/test_conf_change.rs [hunk 1]
```diff
@@ -425,3 +420,40 @@ fn test_server_after_remove_itself() {
...
+    let s2 = (3..4).collect();
+    cluster.partition(Arc::new(s1), Arc::new(s2));
+
+    // during partition, peer 3 don't know it's removed and add again
+    cluster.change_peer(r1, ConfChangeType::RemoveNode, 3);
+    cluster.change_peer(r1, ConfChangeType::AddNode, 3);
+
+    // when network recover, peer 3 will receive heartbeat
+    cluster.reset_transport_hooks();
+    print!("--------------------------------------------\n");
```
Comment (siddontang) [inline tests/raftstore/test_conf_change.rs]: remove any print before pushing to remote.

Comment (tiancaiamao) [inline tests/raftstore/test_conf_change.rs]: I also add init_log() at line 425, which print logs in detail.
this line  "------------" will help us to observe what happens after network restore.

I'll update it later, when we fix the issue and pass this test.

Comment (BusyJay) [inline tests/raftstore/test_conf_change.rs]: `print!` can be replaced with `debug!`.

File: tests/raftstore/test_conf_change.rs [hunk 2]
```diff
@@ -425,3 +420,40 @@ fn test_server_after_remove_itself() {
...
+    sleep_ms(3000);
```
Comment (BusyJay) [inline tests/raftstore/test_conf_change.rs]: I don't think we need to sleep 3 secs here. Maybe we can `reset_leader` then `must_put` to verify raft has panic.

**Comment by tiancaiamao** [comment]
Duplicate https://github.com/pingcap/tikv/pull/574/files

**Pull request closed without merging**